fitbenchmarking/fitting/controllers/minuit_controller.py

```diff
@@ -0,0 +1,72 @@
+"""
+Implements a controller for the CERN pacakage Minuit
+https://seal.web.cern.ch/seal/snapshot/work-packages/mathlibs/minuit/
+using the iminuit python interface
+http://iminuit.readthedocs.org
+"""
+from iminuit import Minuit
+import numpy as np
+
+from fitbenchmarking.fitting.controllers.base_controller import Controller
+
+class MinuitController(Controller):
+    """
+    Controller for the Minuit fitting software
+    """
+
+    def __init__(self, problem, use_errors):
+        """
+        Initializes variable used for temporary storage.
+        """
+        super(MinuitController, self).__init__(problem, use_errors)
+        self._popt = None
+        self._initial_step = None
+
+    def setup(self):
+        """
+        Setup for Minuit
+        """        
+        # minuit requires an initial step size.
+        # The docs say
+        # "A good guess is a fraction of the initial 
+        #  fit parameter value, e.g. 10% 
+        #  (be careful when applying this rule-of-thumb
+        #  when the initial parameter value is zero "
+        self._initial_step = 0.1 * np.array(self.initial_params)
+        # set small steps to something sensible(?)
+        self._initial_step[self._initial_step < 1e-12] = 1e-12
```

Comment:
Is `1e-12` what they suggest as a default tolerance too?

Reply:
No -- it says:

> You can set initital step sizes with keywords, `error_<name> = <step size>`. If you don't know what to pick, don't worry too much. Usually, the choice is not important. In the worst case, iminuit will use a few more function evaluations to find the minimum. A good guess is a fraction of the initial fit parameter value, e.g. 10 % (be careful when applying this rule-of-thumb when your the initial parameter value is zero).

`1e-12` is plucked out of thin air, in the absence of any other information.

Reply:
Maybe once the options file is completed we have a tolerance for things being close to zero

We kinda use it here as well
https://github.com/fitbenchmarking/fitbenchmarking/blob/1c6d59583ce7c7b75922dbed8235998bf4768c83/fitbenchmarking/fitting/controllers/base_controller.py#L74-L75